### PR TITLE
[SD-3028] - Disable packaging widget for read-only packages

### PR DESCRIPTION
**Pull request by akintolga**
Also added initial set of validations for package publishing...

### CODE REVIEW

File: client/app/scripts/superdesk-authoring/widgets/widgets.js [hunk 1]
```diff
@@ -40,7 +40,8 @@ function WidgetsManagerCtrl($scope, $routeParams, authoringWidgets, archiveServi
     });
 
     $scope.isLocked = function(widget) {
-        return widget.needUnlock && $scope.item._locked;
+        return (widget.needUnlock && $scope.item._locked) ||
+        (widget.needUnlock && !$scope.item._editable);
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/widgets/widgets.js]: Any particular reason for `widget.needUnlock` appearing twice?

**Comment by plamut** [comment]
:+1:

**Comment by sivakuna-aap** [comment]
:+1:

**Comment by mdhaman** [comment]
:+1:

**Comment by marwoodandrew** [comment]
:+1:

**Comment by mugurrus** [comment]
:+1:

**Comment by plamut** [comment]
Seems like this PR is among those with the highest number of thumb-ups. :)

**Pull request merged**